internal/js_scanner/js_scanner.go

```diff
@@ -13,24 +13,33 @@ import (
 )
 
 type HoistedScripts struct {
-	Hoisted [][]byte
-	Body    []byte
+	Hoisted     [][]byte
+	HoistedLocs []loc.Loc
+	Body        [][]byte
+	BodyLocs    []loc.Loc
```

Comment:
Previously we were throwing away all location info when hoisting imports/exports, which was obviously unhelpful!